lib/datadog/statsd/sender.rb

```diff
@@ -68,6 +69,9 @@ def add(message)
         # the parent process) and spawn a new companion thread.
         if !sender_thread.alive?
           @mx.synchronize {
+            # an attempt was previously made to start the sender thread but failed.
+            # skipping re-start
+            break if @done
```

Comment:
A question occurred to me on a second pass on this one: is `break` really the intention here, or would `return` be more correct?

Since we're inside a Ruby block, `break` only pops out of the `@mx.synchronize`, and then executes the rest of the method (e.g. from the `if message_queue.length <= @queue_size` on).

Which means we still store all messages whenever there's a failure. Is that intended? (May be worth adding a spec to clarify this -- e.g. that the queue is either empty or not empty after the failure)